### PR TITLE
Painless: Decouple PainlessLookupBuilder and Whitelists

### PR DESCRIPTION
**Pull request by jdconrad**
Mostly a mechanical change.  This moves all code related to the whitelists to a WhitelistsToPainlessLookup class so that PainlessLookupBuilder has no dependencies on the whitelists to build a PainlessLookup.  This allows for Painless to create a Lookup from other sources moving forward which may have practical applications in both testing and jar separation shortly.

### CODE REVIEW

**Comment by jdconrad** [comment]
@rjernst Removed the extraneous class and put whitelistsToPainlessLookup into PainlessLookupBuilder as we discussed.

**Comment by jdconrad** [comment]
@elasticmachine test this please

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/lookup/PainlessLookupBuilder.java [hunk 1]
```diff
@@ -126,14 +126,55 @@ public int hashCode() {
     private static final Pattern METHOD_NAME_PATTERN = Pattern.compile("^[_a-zA-Z][_a-zA-Z0-9]*$");
     private static final Pattern FIELD_NAME_PATTERN  = Pattern.compile("^[_a-zA-Z][_a-zA-Z0-9]*$");
 
-    private final List<Whitelist> whitelists;
+    public static PainlessLookup whitelistsToPainlessLookup(List<Whitelist> whitelists) {
```
Comment (rjernst) [inline modules/lang-painless/src/main/java/org/elasticsearch/painless/lookup/PainlessLookupBuilder.java]: I would name this more simply like `buildFromWhitelist`.

**Review by rjernst — APPROVED** [review]
Thanks, LGTM. Just one minor suggestion.

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/lookup/PainlessLookupBuilder.java (hunk 1 above)
Comment (jdconrad) [reply modules/lang-painless/src/main/java/org/elasticsearch/painless/lookup/PainlessLookupBuilder.java]: Done.

**Comment by jdconrad** [comment]
@rjernst Thanks for the review.  Will commit as soon as CI passes.

**Pull request merged**